src/binder/bind_node_visitor.cpp

```diff
@@ -175,5 +178,38 @@ void BindNodeVisitor::Visit(expression::CaseExpression *expr) {
   }
 }
 
+void BindNodeVisitor::Visit(expression::StarExpression *expr) {
+  if (!BinderContext::HasTables(context_))
+    throw BinderException("Invalid expression" + expr->GetInfo());
+}
+
+// Deduce value type for these expressions
+void BindNodeVisitor::Visit(expression::OperatorExpression *expr) {
+  SqlNodeVisitor::Visit(expr);
+  expr->DeduceExpressionType();
+}
+void BindNodeVisitor::Visit(expression::AggregateExpression *expr) {
+  SqlNodeVisitor::Visit(expr);
+  expr->DeduceExpressionType();
+}
+
+void BindNodeVisitor::Visit(expression::FunctionExpression *expr) {
+  // Visit the subtree first
+  SqlNodeVisitor::Visit(expr);
+
+  // Check catalog and bind function
+  std::vector<type::TypeId> argtypes;
+  for (size_t i = 0; i < expr->GetChildrenSize(); i++)
+    argtypes.push_back(expr->GetChild(i)->GetValueType());
+  // Check and set the function ptr
+  auto catalog = catalog::Catalog::GetInstance();
+  const catalog::FunctionData &func_data =
+      catalog->GetFunction(expr->GetFuncName(), argtypes);
+  LOG_INFO("Function %s found in the catalog", func_data.func_name_.c_str());
```

Comment:
Switch to `LOG_DEBUG`

Reply:
This is not going through LLVM because the ProjectionTranslator requires the projection_plan must have a child. I will add support for that.
The test cases are already there in the optimizer_sql_test.